packages/catalog-process/src/services/validators.ts

```diff
@@ -32,6 +33,36 @@ import {
 } from "../model/domain/errors.js";
 import { ReadModelService } from "./readModelService.js";
 
+export const descriptorStatesAllowingDocumentOperations: DescriptorState[] = [
+  descriptorState.draft,
+  descriptorState.deprecated,
+  descriptorState.published,
+  descriptorState.suspended,
+];
+
+export const updatableDescriptorStates: DescriptorState[] = [
+  descriptorState.deprecated,
+  descriptorState.published,
+  descriptorState.suspended,
+];
+
+export const notActiveDescriptorState: DescriptorState[] = [
+  descriptorState.draft,
+  delegationState.waitingForApproval,
+];
+
+export function isNotActiveDescriptor(descriptor: Descriptor): boolean {
+  return notActiveDescriptorState.includes(descriptor.state);
+}
+
+export function isActiveDescriptor(descriptor: Descriptor): boolean {
+  return !notActiveDescriptorState.includes(descriptor.state);
+}
+
+export function isDescriptorUpdeatable(descriptor: Descriptor): boolean {
```

Comment:
See comment about pattern matching, same here and in general for the validators of the catalog process